engine/rpcserver.go

```diff
@@ -742,7 +742,42 @@ func (s *RPCServer) GetOrders(ctx context.Context, r *gctrpc.GetOrdersRequest) (
 
 // GetOrder returns order information based on exchange and order ID
 func (s *RPCServer) GetOrder(ctx context.Context, r *gctrpc.GetOrderRequest) (*gctrpc.OrderDetails, error) {
-	return &gctrpc.OrderDetails{}, common.ErrNotYetImplemented
+	exch := GetExchangeByName(r.Exchange)
+	if exch == nil {
+		return nil, errors.New("exchange is not loaded/doesn't exist")
+	}
+	result, err := exch.GetOrderInfo(r.OrderId)
+	var trades []*gctrpc.TradeHistory
+	if result.Trades != nil {
+		for i := range result.Trades {
+			trades = append(trades, &gctrpc.TradeHistory{
+				CreationTime: result.Trades[i].Timestamp.Unix(),
+				Id:           result.Trades[i].TID,
+				Price:        result.Trades[i].Price,
+				Amount:       result.Trades[i].Amount,
+				Exchange:     result.Trades[i].Exchange,
+				AssetType:    result.Trades[i].Type.String(),
+				OrderSide:    result.Trades[i].Side.String(),
+				Fee:          result.Trades[i].Fee,
+			})
+		}
+	}
+	return &gctrpc.OrderDetails{
+		Exchange:      result.Exchange,
+		Id:            result.ID,
+		BaseCurrency:  result.CurrencyPair.Base.String(),
+		QuoteCurrency: result.CurrencyPair.Quote.String(),
+		AssetType:     asset.Spot.String(),
```

Comment:
This is not always the case, should be result.Asset.

Reply:
true, but I don't see an `Asset` neither in `order.Detail `nor in `GeneralizedOrderResponse`

Reply:
I've added `asset` to `order.detail` in a pending PR, so this could get updated after that point. You could call `exch.GetPairAssetType(result.CurrencyPair)`

Reply:
ok, I added a TODO on my code. Could you tell me the PR number so I can track it?

Reply:
#446

Reply:
446 has been merged now!